src/shared/search/GlobalSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {FC, useEffect, memo, useState, useMemo, useCallback} from 'react'
+import {useLocation} from 'react-router-dom'
+import {DocSearchModal} from '@docsearch/react'
+import {v4 as uuid} from 'uuid'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import '@docsearch/react/style'
+
+import './GlobalSearch.scss'
+
+import {event} from 'src/cloud/utils/reporting'
+
+const Hit: FC<any> = memo(({hit, children}) => {
```

Comment:
should this be `useMemo`?

Reply:
no, useMemo is to memoize a value, memo is to memoize a React component